syntax = "proto3";

package dfs_service;

service DFSService {
    rpc StoreFile (stream StoreChunk) returns (StoreResponse);
    rpc FetchFile (FetchRequest) returns (stream FetchChunk);
    rpc DeleteFile (DeleteRequest) returns (DeleteResponse);

    // 4. REQUIRED (Parts 1 & 2): A method to list all files on the server

    // 5. REQUIRED (Parts 1 & 2): A method to get the status of a file on the server

    // 6. Any other methods you deem necessary to complete the tasks of this assignment

}
// Data Chunk for store operation
message StoreChunk {
    string filename = 1;
    bytes data = 2;
}
// Response for store operation
message StoreResponse {
    string filename = 1;
    int64 mtime = 2;
}

// Request for fetch operation
message FetchRequest {
    string filename = 1;
}

// Data Chunk for fetch operation
message FetchChunk {
    string filename = 1;
    int64 mtime = 2;
    bytes data = 3;
}

// Request for delete operation
message DeleteRequest {
    string filename = 1;
}

// Response for delete operation
message DeleteResponse {
    string filename = 1;
}